app/models/post.rb

```diff
@@ -0,0 +1,16 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :likes, dependent: :destroy
+  has_many :comments, dependent: :destroy
+
+  # update the posts counter for a user by 1
+  def update_posts_counter(user)
+    user.posts_counter = user.posts_counter.to_i + 1
+    user.save!
+  end
```

Comment:
- [ Required ] I think it would be nice if you use the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_posts_counter` method every time a new `Post` has been saved to the database, then make the update_posts_counter method private. This will ensure that the posts counter is incremented when a new post has been saved to the database, please see the suggestion below for an idea of how to achieve this.

## Example code
```ruby
class Post < ApplicationRecord
  
  after_save :update_post_counter

  private

  def update_posts_counter
     author.increment(:posts_counter)
  end
end
```